dali/c_api_2/data_objects_test.cc

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

Comment:
Are you testing non-contiguous tensor lists here?

Reply:
I'm testing `daliTensorListAttachBuffer` with custom offsets.
The test for `daliTensorListAttachSamples` is missing. I'll add.

Reply:
Done.